portal-impl/src/com/liferay/counter/service/persistence/impl/CounterFinderImpl.java

```diff
@@ -118,7 +119,21 @@ public long increment(String name, int size) {
 
 	@Override
 	public void invalidate() {
-		_counterRegisterMap.clear();
+		if (!DBPartition.isPartitionEnabled() ||
+			(CompanyThreadLocal.getCompanyId() == CompanyConstants.SYSTEM)) {
+
+			_counterRegisterMap.clear();
+
+			return;
+		}
+
+		for (String key : _counterRegisterMap.keySet()) {
+			if (key.endsWith(
+					StringPool.AT + CompanyThreadLocal.getCompanyId())) {
+
+				_counterRegisterMap.remove(key);
```

Comment:
@marianoalvarosaiz This could cause ConcurrentModificationException, please use iterator

Reply:
@tinatian _counterRegisterMap is a ConcurrentHashMap, so it's ConcurrentModificationException , see https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#keySet--